src/components/molecules/SecretPasswordForm/SecretPasswordForm.tsx

```diff
@@ -1,34 +1,51 @@
 import React, { useCallback, useState } from "react";
-import { useHistory } from "react-router-dom";
 
 import { checkAccess } from "api/auth";
 
-import { venueEntranceUrl } from "utils/url";
 import { setLocalStorageToken } from "utils/localStorage";
 import { isTruthy } from "utils/types";
 
 import { useVenueId } from "hooks/useVenueId";
+import { useUser } from "hooks/useUser";
 
 import "./SecretPasswordForm.scss";
 
-const SecretPasswordForm = ({ buttonText = "Join the party" }) => {
-  const history = useHistory();
+interface SecretPasswordFormProps {
+  buttonText: string;
+  onPasswordSubmit?: () => void;
+  onPasswordSuccess?: () => void;
+}
+
+export const SecretPasswordForm: React.FC<SecretPasswordFormProps> = ({
+  buttonText = "Join the party",
+  onPasswordSubmit,
+  onPasswordSuccess,
+}) => {
+  const { user } = useUser();
   const venueId = useVenueId();
 
-  const [error, setError] = useState();
-  const [password, setPassword] = useState();
-  const [message, setMessage] = useState();
+  const [error, setError] = useState(false);
+  const [password, setPassword] = useState("");
+  const [message, setMessage] = useState("");
 
-  function passwordChanged(e) {
+  const passwordChanged = (e: React.ChangeEvent<HTMLInputElement>) => {
     setPassword(e.target.value);
     setMessage("");
     setError(false);
-  }
+  };
 
   const passwordSubmitted = useCallback(
     async (e) => {
+      if (!venueId) return;
+
       e.preventDefault();
```

Comment:
If venueId doesn't exist this component won't render at all, because of the guards in the higher hierarchy components. In this case I don't think it matters that much, but I guess we want to preventDefault only if there is some logic being executed. In this case only when venueId exists.

Reply:
_nods_ cool, makes sense